src/gc.c

```diff
@@ -344,6 +345,16 @@ static void run_finalizers(void)
     JL_GC_POP();
 }
 
+void jl_gc_inhibit_finalizers(int state)
+{
+    if (jl_gc_finalizers_inhibited && !state && !jl_in_gc) {
+        jl_in_gc = 1;
```

Comment:
Remind me why do we need to set `jl_in_gc` when running finalizers? There doesn't seem to be a problem to enable GC since the arguments are rooted. Is this just for consistency for not allowing task switch in them?

I'm asking mainly because requiring no GC during finalizer runs doesn't look like a trival synchronization to do.

c.c. @carnaval